go.mod

```diff
@@ -54,7 +54,6 @@ require (
 	github.com/mattn/go-colorable v0.1.8 // indirect
 	github.com/mattn/go-isatty v0.0.13 // indirect
 	github.com/mattn/go-runewidth v0.0.13 // indirect
-	github.com/mattn/go-sqlite3 v1.14.7 // indirect
```

Comment:
any reason to use modernc.org/sqlite instead of github.com/mattn/go-sqlite3?

Reply:
It compiles with `CGO_ENABLED=0`. See also https://github.com/temporalio/temporal/pull/1968#discussion_r715118938 and https://github.com/temporalio/temporal/pull/1968#discussion_r715241946

Reply:
`modernc.org/sqlite` seems to introduce lots of new dependencies, can we use existing `github.com/mattn/go-sqlite3`? 
what features are supported by `modernc` but not existing one?